scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -0,0 +1,1226 @@
+
+            @param access_time: The time of the access.
+            @type access_time: double
+            """
+            self.__last_access_time = access_time
+
+
+class KubernetesMonitor( ScalyrMonitor ):
+    """Monitor plugin for kubernetes
+
+    This plugin is based of the docker_monitor plugin, and uses the raw logs mode of the docker
+    plugin to send kubernetes logs to Scalyr.  It also reads labels from the Kubernetes api and
+    associates them with the appropriate logs.
+    TODO:  Back fill the instructions here.
+    """
+
+    def __get_socket_file( self ):
+        """Gets the Docker API socket file and validates that it is a UNIX socket
+        """
+        #make sure the API socket exists and is a valid socket
+        api_socket = self._config.get( 'api_socket' )
+        try:
+            st = os.stat( api_socket )
+            if not stat.S_ISSOCK( st.st_mode ):
+                raise Exception()
+        except:
+            raise Exception( "The file '%s' specified by the 'api_socket' configuration option does not exist or is not a socket.\n\tPlease make sure you have mapped the docker socket from the host to this container using the -v parameter.\n\tNote: Due to problems Docker has mapping symbolic links, you should specify the final file and not a path that contains a symbolic link, e.g. map /run/docker.sock rather than /var/run/docker.sock as on many unices /var/run is a symbolic link to the /run directory." % api_socket )
+
+        return api_socket
+
+    def _initialize( self ):
+        data_path = ""
+        log_path = ""
+        host_hostname = ""
+
+        if self._global_config:
+            data_path = self._global_config.agent_data_path
+            log_path = self._global_config.agent_log_path
+
+            if self._global_config.server_attributes:
+                if 'serverHost' in self._global_config.server_attributes:
+                    host_hostname = self._global_config.server_attributes['serverHost']
+                else:
+                    self._logger.info( "no server host in server attributes" )
+            else:
+                self._logger.info( "no server attributes in global config" )
+
+        self.__socket_file = self.__get_socket_file()
+        self.__docker_api_version = self._config.get( 'docker_api_version' )
+
+        self.__client = DockerClient( base_url=('unix:/%s'%self.__socket_file), version=self.__docker_api_version )
+
+        self.__glob_list = self._config.get( 'container_globs' )
+
+        self.__report_container_metrics = self._config.get('report_container_metrics')
+
+        self.__container_checker = None
+        if self._config.get('log_mode') != 'syslog':
+            self.__container_checker = ContainerChecker( self._config, self._logger, self.__socket_file, self.__docker_api_version, host_hostname, data_path, log_path )
+
+        self.__network_metrics = self.__build_metric_dict( 'docker.net.', [
```

Comment:
Same question here.. is it still appropriate to call them `docker` metrics?  Now... for our customers, maybe this is easier because they could have dashboards built using these metric names.  What do you think?  Do people generally think of this as not running docker?

Reply:
As mentioned via email, I think it's still worth calling them docker metrics, because they are metrics for each docker container, and there are likely separate k8s specific metrics we could be collecting too, so calling the docker metrics 'docker_metrics' helps differentiate.

Reply:
That sounds fine to me.  You are more in touch with what the community will expect.